### PR TITLE
Fix application default credentials

### PR DESCRIPTION
**Pull request by thclark**
## Summary

Provides progress on #374 by importing default credentials from compute engine instances.

### Fixes
- Add read of default account credentials on compute engine instances
- Correctly pull ADCs using octue credentials manager

### Operations
- Ignore IDE files
- Add a .devcontainer for vscode users
- Ignore IDE files
- Add a .devcontainer for vscode users
- Update version
- Ensure gcp credentials aren't baked into dockeer or committed
- Revert to python 3.8 for developing octue in vscode

### Dependencies
- Add pytest to dev dependencies

### Testing
- Update unit test for new credentials

### Chores
- Apply style to the CI file
- Apply style to the CI file

### Other
- Remote branch

### CODE REVIEW

**Comment by thclark** [comment]
@cortadocodes I've had to merge this quickly, because all our services are down but please do review the code.

File: octue/cloud/credentials.py [hunk 1]
```diff
@@ -1,61 +1,106 @@
...
     :return None:
     """
 
-    def __init__(self, environment_variable_name="GOOGLE_APPLICATION_CREDENTIALS"):
+    def __init__(self, environment_variable_name=DEFAULT_ENVIRONMENT_VARIABLE_NAME):
+
+        if environment_variable_name is None:
+            raise InvalidInputException(
+                "Cannot accept `None` as an environment variable name. Either specify the name as a string or use the default value"
+            )
```
Comment (cortadocodes) [inline octue/cloud/credentials.py:29]: Why can't it be `None` here? I added the option for a specific nuanced use case but can't remember exactly what it was

File: octue/cloud/credentials.py [hunk 2]
```diff
@@ -1,61 +1,106 @@
...
+
         self.environment_variable_name = environment_variable_name
 
-        if self.environment_variable_name is None:
-            self.service_account_json = None
-            return
+        # Note: We use '' rather than None to allow us to patch the value in testing (you cannot correctly unset variables in a patch of os.environ)
+        self.environment_variable_value = os.getenv(environment_variable_name, "")
```
Comment (cortadocodes) [inline octue/cloud/credentials.py:34]: Does
```
with mock.patch.dict(os.environ, clear=True):
```
not do it?

File: tests/cloud/test_credentials.py [hunk 3]
```diff
@@ -43,6 +51,9 @@ def test_credentials_can_be_loaded_as_dictionary_from_file(self):
                 credentials = GCPCredentialsManager().get_credentials(as_dict=True)
                 self.assertEqual(credentials, {"blah": "nah"})
 
-    def test_credentials_are_none_when_environment_variable_name_is_none(self):
-        """Test that credentials are `None` when the given environment variable name is `None`."""
-        self.assertIsNone(GCPCredentialsManager(environment_variable_name=None).get_credentials())
+    def test_compute_engine_credentials_are_used_when_environment_unset(self):
+        """Test that application default credentials are accessed when the environment variable value is not set."""
+        with patch.dict(os.environ, {"GOOGLE_APPLICATION_CREDENTIALS": ""}):
```
Comment (cortadocodes) [inline tests/cloud/test_credentials.py:56]: Yeah I think you can use this here:
```
with mock.patch.dict(os.environ, clear=True):
```

**Pull request merged**